app/main.py

```diff
@@ -1,25 +1,73 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, value: str) -> None:
+        self.protected_name = "_" + value
+
+    def __get__(self, instance: Any, owner: Any) -> Any:
```

Comment:
same about instance. Fix everywhere

Reply:
updated, though I'm not sure what type I should use since descriptors should handle any type